src/functions/durableOrchestration.ts

```diff
@@ -0,0 +1,32 @@
+import { HttpRequest, InvocationContext, output, trigger } from '@azure/functions';
+import * as df from 'durable-functions';
+import { ActivityHandler, DurableClientHandler, OrchestrationHandler } from 'durable-functions';
+
+// Replace with the name of your Durable Functions Activity
+const activityName = 'hello';
+
+const orchestrator: OrchestrationHandler = function* (context) {
+    const outputs = [];
+    outputs.push(yield context.df.callActivity(activityName, 'Tokyo'));
+    outputs.push(yield context.df.callActivity(activityName, 'Seattle'));
+    outputs.push(yield context.df.callActivity(activityName, 'Cairo'));
+
+    return outputs;
+};
+df.orchestration('durableOrchestrator1', orchestrator);
+
+const helloActivity: ActivityHandler<string> = (_context: InvocationContext, input: string) => {
+    return `Hello, ${input}`;
+};
+df.activity<string>(activityName, { handler: helloActivity });
+
+const clientHandler: DurableClientHandler = async (context: InvocationContext, request: HttpRequest, client) => {
+    const instanceId = await client.startNew(request.params.orchestratorName, undefined, request.text());
+    context.log(`Started orchestration with ID = '${instanceId}'.`);
+    return client.createCheckStatusResponse(request, instanceId);
+};
+df.client('durableOrchestrationStart1', {
```

Comment:
The code you just wrote is still valid -- there would be no further changes required from the SDK side to enable that workflow. Are you talking more from the point of view of which pattern do we recommend in our samples?

 > There might be a way to get the benefits of both worlds

That's exactly what I was trying to do with `df.httpClient()` 😭 Perhaps we can bring it back but also take an object instead of the method directly, therefore setting less defaults and enabling more customization? It would just behave the same as `app.http()` essentially, except taking in a `DurableClientHandler` as the handler

Reply:
> Are you talking more from the point of view of which pattern do we recommend in our samples?

I'm saying we should remove the `df.client` method until we can redesign it to have both trigger and client intellisense.  Until we redesign it, let's just recommend the above pattern that doesn't require any changes from the sdk

> That's exactly what I was trying to do with df.httpClient() 😭 Perhaps we can bring it back

Regardless, it should be handled outside this PR. We would need to file an issue, decide if it's worth it (the above experience might be just fine), design it, and lastly implement it.

Also, I don't think `df.httpClient` as-is would've worked. For one reason, orchestration clients are used with many different triggers and I'd prefer if we found a way to generically support all triggers without having to create a new `df.httpClient` and `df.timerClient` method for each.

Reply:
That's fair, except there's one other notable exception about HTTP triggers, besides it being the most used trigger type, which is that it has a primary HTTP output binding configured by default, which many users would not think to do, especially if they use the new programming model where that is already configured for them. Basically, if you want to use a client method with a timer trigger using only `df.client()`, all you really lose out on is intellisense, whereas for HTTP triggers, you lose _both_ intellisense _and_ the automatic HTTP output binding.

It sounds like we have different goals/priorities too, because for me, I'd rather get all the APIs we want to support out there as soon as possible (even if they aren't perfect), in a public version of the Durable SDK, so that we can send it out to partners and have internal testing and later on the public preview, and so on, to capitalize on the feedback we get from that to inform our decisions to improve/update/even remove certain APIs, rather than miss out on that rich period of feedback.

Reply:
From what I understand: the convenience of `df.client` is in that it makes DF Client into a proper parameter of the Function.

I think that's really valuable and it would match what C# does today as well as what PyStein will do for DF. In both - DF Clients are proper parameters.

**However**, I was reviewing the wiki for the new Node programming model and saw [here](https://github.com/Azure/azure-functions-nodejs-library/wiki/Secondary-inputs-and-outputs) that 2ndary inputs (like DF Clients) appear to **never** be a proper parameter. Instead, it seems they need to be manually retrieved from `context.extraInputs`. Is that accurate?

_If so_, I would not want to deviate from that pattern at this time. Put differently - I would support utilizing just `app.http` as Eric suggested.

In the case of Python, not providing a fully constructed DF Client would have deviated from what their programming model does in every other case, which is why we invested heavily in making sure a DF Client could be made into a proper parameter. For Node, it seems the opposite is true at this time.

Reply:
I think there's definitely room for the DF SDK to "do its own thing" and eventually implement a feature like `df.httpClient`. In fact, other members of the DF team suggested this almost as soon as the new programming models were introduced. Hossam is right in that, overwhelmingly, DF Clients leverage HTTP Triggers. But I also think that adding those new features should not be in the scope of this particular PR. For now, I would want us to focus on the minimum viable experience, to ensure we deliver this in a timely manner.

Reply:
> It sounds like we have different goals/priorities too, because for me, I'd rather get all the APIs we want to support out there as soon as possible (even if they aren't perfect)

Believe it or not, my goal is also to get the most stuff merged as soon as possible. The way to do that is to have smaller targeted PRs for each API.

> Hossam is right in that, overwhelmingly, DF Clients leverage HTTP Triggers.

I shared a query offline that http was _not_ overwhelming. I'm curious if you have an alternate query showing otherwise.

Reply:
> I shared a query offline that http was not overwhelming. I'm curious if you have an alternate query showing otherwise.

@ejizba : I don't have one handy. I can (and should) definitely look to validate that with data but I already agree that a `df.httpClient` API would be better suited for a follow-up PR. Now, this depends on my current understanding that this new prog-model always requires users to manually extra secondary input bindings (like the DF client) from the `extraInputs` property. Is that correct, or do you already have examples of secondary input bindings that are injected as proper paramteters?

Reply:
Your understanding is correct. Durable would be the first to inject a secondary input as a proper parameter.

Reply:
So it looks like here we are leaning towards removing `df.client()` for the purposes of this PR in favor of using `getClient()` as Eric showed above. I went ahead and did that and updated the SDK PR and this PR with these changes.